docs/pip.md

```diff
@@ -0,0 +1,162 @@
+# Pip
+
+This document describes some of the more intricate details of Cachito support for pip. For a high
+level overview, look [here in the README](../README.md#pip).
+
+Cachito has a number of specific requirements when it comes to pip packages. Some of those stem from
+the general ideas behind Cachito (e.g. reproducibility), some from the technical challenges of
+supporting a packaging system which defines most metadata through a Python executable. Read on for
+more details.
+
+## requirements.txt
+
+One of the main component of a pip package is the `requirements.txt` file. Typically, the file might
+look something like this:
+
+```ini
+requests
+git+https://github.com/containerbuildsystem/dockerfile-parse
+https://github.com/containerbuildsystem/operator-manifest/archive/v0.0.3.zip
+```
+
+The dependencies in this file are:
+
+* `requests`, a PyPI dependency
+* `dockerfile-parse`, a Git dependency
+* `operator-manifest`, an HTTPS dependency
+
+Git and HTTP(S) dependencies will henceforth collectively be referred to as "external."
+
+### Pinning versions
+
+To make sure builds are reproducible, Cachito will require that all dependencies be pinned to
+a specific version.
+
+* for PyPI dependencies, use the `==` operator: `requests==2.24.0`
+* for Git dependencies, specify the commit hash:
+  * `git+https://github.com/containerbuildsystem/dockerfile-parse@ed38dc2eadf6890b524340f45a7d172d02196e61`
+* for HTTP(S) dependencies, include the hash of the source archive (truncated for readability):
+  * `https://github.com/containerbuildsystem/operator-manifest/archive/v0.0.3.zip#cachito_hash=sha256:7774db0`
+
+### Recursive dependencies
+
+You will need to include not only the direct dependencies for your application, but all the
+dependencies of those dependencies (recursively) as well. There are two reasons for this. The first
+one is reproducibility (you will need to pin the versions of recursive dependencies as well). The
+second reason is that, to determine the dependencies of your dependencies, Cachito would have to
+execute their `setup.py` files, which is not desirable.
+
+### pip-compile
+
+Good news, there is a tool which does part of the required setup for you. That tool is `pip-compile`
+from the [pip-tools](https://github.com/jazzband/pip-tools) Python package. Below is an example of
+how you would use it.
+
+1. rename your `requirements.txt` to `requirements.in` (convention)
+2. run `pip-compile requirements.in -o requirements.txt`
```

Comment:
I think that this can be simplified a little bit.. maybe something like

### Specifying dependencies

In addition to specifying direct dependencies, recursive dependencies also need to be explicitly defined for two reasons:

1. Further enable reproducibility by explicitly specifying _every_ needed package
2. Prevent the need for remote execution of `setup.py`

#### pip-compile

While this might be onerous to manually maintain, `pip-compile` from [pip-tools](https://github.com/jazzband/pip-tools) can be used to automate this process for you using the following procedure.

1. rename `requirements.txt` to `requirements.in` (by convention)
2. run `pip-compile requirements.in -o requirements.txt`

Reply:
If you want to put more information into the first 1/2, you can. I _assume_ that most people understand how remote execution of code is generally not desired, but if that is not  widely known, feel free to add more there.

Reply:
Thanks, I used your version :-)